### ISSUE TITLE
[web] a11y focus ring rect frequently broken for plain text

### ISSUE THREAD

**Issue by yjbanov** [flutter/flutter#146774]
Due to https://g-issues.chromium.org/issues/40875151 (I filed it against Chromium, but all browsers have it; it's a missing piece in the ARIA spec), I've been having difficulty finding a combination of DOM structure and ARIA attributes for plain text that would satisfy all screen readers. Currently, text is rendered into semantics like this:

```
<flt-semantics style="position: absolute;
                      width: 814px;
                      height: 809px;
                      top: 0px;
                      left: 0px;
                      pointer-events: all;">
  Hello, World!
</flt-semantics>
```

All screen readers are able to find this text and read it to the user correctly. However, some screen reader + browser combos, e.g. VoiceOver on macOS, ignore the size information specified on the element, and instead tightly hug the text that's inside the element:

![Screenshot 2024-04-15 at 10 59 41 AM](https://github.com/flutter/flutter/assets/211513/23988cc3-85f4-40c2-8e6d-2ce7a454142f)

The framework means the box containing the text to cover the whole screen with the "Hello, World!" text centered horizontally inside it. However, the DOM used to report semantics to screen readers is unaware of all this. All it gets is the rect of the box `(0, 0, 814, 809)` and the label ("Hello, World!"), and there does not seem to be a way to publish such box via anything DOM and ARIA provide.

Repro:

```dart
import 'package:flutter/material.dart';
import 'package:flutter/semantics.dart';

void main() {
  runApp(const MyApp());
  SemanticsBinding.instance.ensureSemantics();
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Plain text accessibility',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Text('Hello, World!', textAlign: TextAlign.center),
    );
  }
}
```

## Methods tried before

* Set `aria-label` and no DOM text inside:
  * Breaks JAWS (and JAWS [refuses to fix it](https://github.com/FreedomScientific/standards-support/issues/759)).
* `role="text"`
  * Supported by Safari <16 (and used to work perfectly), but alas, [Safari 17 removed it](https://github.com/w3c/aria/issues/2011).
* Other roles: "group", "static"
  * Sizes the ring correctly but breaks the traversal with "end of $LABEL, grou" wording after every text element.
* Adding hidden content inside the node (to avoid "empty group")
  * Hidden content gets collapsed as if the node never had it, so it has no effect.

## Links

* Various discussions of `role="text"` removal:
  * https://github.com/w3c/aria/issues/2011
  * https://bugs.webkit.org/show_bug.cgi?id=260641
  * https://github.com/w3c/aria/issues/870
* JAWS:
  * https://github.com/flutter/flutter/issues/122607
  * https://github.com/FreedomScientific/standards-support/issues/759
* Internal Google issues:
  * http://b/334798120
  * http://yaqs/8749427549701931008

**Comment by yjbanov** [comment]
I think I found a trick: measure the DOM text then rescale it to match the `rect` of the semantics node. Here's a sketch:

```
  <div style="width: 200px; height: 100px; border: 1px solid black;">
    <span id="transformedSpan" style="display: inline-block; transform-origin: 0 0 0">Hello, World!</span>
  </div>
  <script>
    let span = document.querySelector('#transformedSpan');
    let rect = span.getBoundingClientRect();
    let scaleX = 200 / rect.width;
    let scaleY = 100 / rect.height;
    span.style.transform = `scale(${scaleX}, ${scaleY})`;
  </script>
```

I tried it on macOS with all the browsers and it worked. I'll try to implement it in the engine.
